canonical-web-and-design/canonical.com
# Add application page for candidates

## Done

- This is a page dedicated to applications for candidates that apply to jobs. This link will be shared in the emails exchanged between the recruiter and the candidate
- A candidate can have multiple pages if the candidate applies to multiple jobs
- This page will let the candidate know what is the progress of his/her application
- The page shows a list of useful information like:
  - Job link
  - Next interview (with a link to convert the time to your own timezone)
  - Previous interviews
  - What stage is the application (WIP)
  - Some useful links about Canonical (WIP)
  - A form to delete the candidate application (WIP)
- The page is not indexed 
- The link is made with `first_name`, `candidate_id` and `application_id`. This is made to avoid someone doing a bruteforce to find all those links: `/careers/application/SUPER-ENCRYPTED-LINK`
- Only the first name of the candidate is displayed on purpose in case someone finds the link 

Here is a copydocs with the proposed content: https://docs.google.com/document/d/1R3-aIHtdrs0eD0qFI8LiUPOf-fVY6IoxAsdjpd-2WO4/edit#heading=h.4ir3ufdci83f

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: http://0.0.0.0:8002/
- Test those URLs (to avoid putting any public link the URLs are in the pastebin) https://pastebin.canonical.com/p/f49wrH9mHM/

# TODO

- [x] A cool design
- [x]  What stage is the application 
- [x] A form to delete the candidate application (Can be done in another PR)
- [x] Some useful links about Canonical
- [x] Update link to be: a unique UUID, this will come with a script that should update all the candidates to add the extra variable in greenhouse.
